chia/consensus/blockchain.py

```diff
@@ -296,8 +296,8 @@ async def receive_block(
                 )
                 raise
 
-            # This is done outside the try-except in case it fails, since we do not want to revert anything if it does
-            await self.__height_map.maybe_flush()
```

Comment:
I moved this out of the `write_db()` block. There's no point in keeping the DB write lock while flushing the height map.